File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/mad/Java/Python/s041542819_commented.py
class LightScanner:
    def __init__(self):
        self.tokens = []
        self.position = 0

    def next_token(self):
        if self.position == len(self.tokens):
            self.tokens = input().split()
            self.position = 0
        token = self.tokens[self.position]
        self.position += 1
        return token

    def next_int(self):
        return int(self.next_token())

    def next_long(self):
        return int(self.next_token())

def lower_bound(arr, target, min_idx=0):
    max_idx = len(arr)
    while min_idx < max_idx:
        mid = (min_idx + max_idx) // 2
        if target <= arr[mid]:
            max_idx = mid
        else:
            min_idx = mid + 1
    return min_idx

def main():
    scanner = LightScanner()
    a = scanner.next_int()
    b = scanner.next_int()
    q = scanner.next_int()

    s = [0] * (a + 2)
    s[0] = -10**10
    for i in range(a):
        s[i + 1] = scanner.next_long()
    s[a + 1] = 20000000000

    t = [0] * (b + 2)
    t[0] = -10**10
    for i in range(b):
        t[i + 1] = scanner.next_long()
    t[b + 1] = 20000000000

    for _ in range(q):
        x = scanner.next_long()
        idx_s = lower_bound(s, x + 1)
        prev_s_idx = max(0, idx_s - 1)
        sl = x - s[prev_s_idx]

        idx_t = lower_bound(t, x + 1)
        prev_t_idx = max(0, idx_t - 1)
        tl = x - t[prev_t_idx]

        next_s_idx = lower_bound(s, x)
        sr = s[next_s_idx] - x

        next_t_idx = lower_bound(t, x)
        tr = t[next_t_idx] - x

        case1 = max(sl, tl)
        case2 = max(sr, tr)
        case3 = 2 * sl + tr
        case4 = 2 * tl + sr
        case5 = sl + 2 * tr
        case6 = tl + 2 * sr

        ans = min(case1, case2, case3, case4, case5, case6)
        print(ans)

if __name__ == "__main__":
    main()
# End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/inline/comment/direct/Java/C/s041542819_commented.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define MAXN 100005

typedef long long ll;

ll s[MAXN], t[MAXN];

ll min(ll a, ll b) {
    return a < b ? a : b;
}

ll abs(ll x) {
    return x > 0 ? x : -x;
}

ll lower_bound(ll *a, ll n, ll t) {
    int min = 0, max = n;
    while (min < max) {
        int mid = (min + max) / 2;
        if (t <= a[mid]) {
            max = mid;
        } else {
            min = mid + 1;
        }
    }
    return min;
}

void solve(int a, int b, int q) {
    for (int i = 0; i < q; i++) {
        ll x;
        scanf("%lld", &x);
        ll idx_s = lower_bound(s, a + 2, x);
        ll idx_t = lower_bound(t, b + 2, x);
        ll sl = abs(x - s[idx_s - 1]);
        ll sr = abs(s[idx_s] - x);
        ll tl = abs(x - t[idx_t - 1]);
        ll tr = abs(t[idx_t] - x);
        ll ans = min(min(max(sl, tl), max(sr, tr)),
                     min(2 * sl + tr, 2 * tl + sr));
        ans = min(ans, min(sl + 2 * tr, tl + 2 * sr));
        printf("%lld\n", ans);
    }
}

int main() {
    int a, b, q;
    scanf("%d %d %d", &a, &b, &q);

    s[0] = t[0] = -10000000000LL;
    for (int i = 1; i <= a; i++) {
        scanf("%lld", &s[i]);
    }
    for (int i = 1; i <= b; i++) {
        scanf("%lld", &t[i]);
    }
    s[a + 1] = t[b + 1] = 20000000000LL;

    solve(a, b, q);

    return 0;
}
// End of Code
